### PR TITLE
Add plop migration generator

### PR DESCRIPTION
**Pull request by aaronccasanova**
### WHY are these changes introduced?

Streamlines the process of creating new SCSS and TypeScript migrations

### WHAT is this pull request doing?

This PR introduces [plop](https://www.npmjs.com/package/plop) as a dev dependency to the Polaris migrator for generating SCSS and TypeScript migrations. Check out this branch and enter the following commands to run the `plop` generator, answer the prompts, and scaffold a new migration:

```sh
yarn && yarn build -- --filter=@shopify/polaris-migrator && cd polaris-migrator && yarn generate
```

See [plop](https://www.npmjs.com/package/plop) and [plop-dir](https://www.npmjs.com/package/plop-dir) to learn more about directory based templating with `plop`

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by sam-b-rose** [comment]
Looks good so far! I think we just need tests for the TypeScript template as well 💯

File: polaris-migrator/plopfile.mjs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      templateDir: path.join(__dirname, './templates/sass-migration'),
+      outputDir: path.join(__dirname, './src/migrations'),
+      prompts: [
+        {
+          name: 'migrationName',
+          message: 'Name of the migration',
+          suffix: ' (e.g. replace-sass-layout)',
+        },
+        {
+          name: 'testFixtureName',
```
Comment (sam-b-rose) [inline polaris-migrator/plopfile.mjs]: Should we just use the `migrationName` to also name the fixture? Seems like that has been our current pattern and  I don't see why we'd need to have a separate `testFixtureName` unless there are multiple input/output tests.

File: polaris-migrator/templates/sass-migration/{{kebabCase migrationName}}/tests/{{kebabCase testFixtureName}}.input.scss.hbs [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.root {
```
Comment (sam-b-rose) [inline polaris-migrator/templates/sass-migration/{{kebabCase migrationName}}/tests/{{kebabCase testFixtureName}}.input.scss.hbs]: `.root` or `:root`? Is this a confusing migration name or does it not really matter? Maybe just call it `.my-class` or `.className`

File: polaris-migrator/templates/sass-migration/{{kebabCase migrationName}}/{{kebabCase migrationName}}.ts.hbs [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import type {FileInfo} from 'jscodeshift';
+import postcss, {Plugin} from 'postcss';
+import valueParser from 'postcss-value-parser';
+
+const plugin = (): Plugin => ({
+  postcssPlugin: '{{kebabCase migrationName}}',
+  Declaration(decl) {
+    // const prop = decl.prop;
```
Comment (sam-b-rose) [inline polaris-migrator/templates/sass-migration/{{kebabCase migrationName}}/{{kebabCase migrationName}}.ts.hbs:8]: Should we uncomment or remove this?

**Review by sam-b-rose — APPROVED** [review]
Looks great, tested and verified. For the case where a user uses an existing migration name (ex: `replace-sass-spacing`), is there a way to validate this input before generating? Maybe we can leverage the [`validate` option](https://github.com/SBoudrias/Inquirer.js/#objects) in `setGenerator`

File: polaris-migrator/plopfile.mjs (hunk 1 above)
Comment (aaronccasanova) [reply polaris-migrator/plopfile.mjs]: [Absolutely](https://github.com/Shopify/polaris/pull/7218/commits/c57efebab063da71538141514333d5e59efe7529) 👍 I modeled everything off the [replace-sass-spacing migration](https://github.com/Shopify/polaris/tree/2fec9611dec0a147be0be0410e955e92c025edc0/polaris-migrator/src/migrations/replace-sass-spacing/tests), but I much prefer the simplified `plop-dir` config!

File: polaris-migrator/templates/sass-migration/{{kebabCase migrationName}}/tests/{{kebabCase testFixtureName}}.input.scss.hbs (hunk 2 above)
Comment (aaronccasanova) [reply polaris-migrator/templates/sass-migration/{{kebabCase migrationName}}/tests/{{kebabCase testFixtureName}}.input.scss.hbs]: Good shout! I like the clarity of `.my-class`: https://github.com/Shopify/polaris/pull/7218/commits/fe16a354e1b686fae23c281be350da605671aede

File: polaris-migrator/templates/sass-migration/{{kebabCase migrationName}}/{{kebabCase migrationName}}.ts.hbs (hunk 3 above)
Comment (aaronccasanova) [reply polaris-migrator/templates/sass-migration/{{kebabCase migrationName}}/{{kebabCase migrationName}}.ts.hbs:8]: I have no preference and will defer to you! I left it in (commented) as `decl.prop` is commonly accessed in PostCSS plugins, but fails `yarn build` following `yarn generate sass-migration` (when uncommented)

**Comment by aaronccasanova** [comment]
> is there a way to validate this input before generating? Maybe we can leverage the [validate option](https://github.com/SBoudrias/Inquirer.js/#objects) in setGenerator

Great callout @samrose3! I think I landed on a pretty elegant solution: https://github.com/Shopify/polaris/pull/7218/commits/75fb06aed68e194043360f944ef12cb3a579e326

Screen Shot 2022-09-20 at 10 47 09 AM

**Pull request merged**